### DISCUSSION TITLE
spamd_url_init: can we pre-compile TLD suffixes?

### DISCUSSION THREAD

**Discussion by Savageman** [rspamd/rspamd#3984] [Q&A] [answered]
Hello there,

When the daemon starts, I have this rspamd_url_init taking a lot of time during initialization to "start compiling of 9628 TLD suffixes; it might take a long time".

Is there a way to pre-compile them in advance so the 1st run is faster?

Thanks in advance for all your kindness :)

**Comment by vstakhov** [comment, marked as answer]
This is now implemented via https://github.com/rspamd/rspamd/commit/81ba9f425ad51141c2f9fe1213fd839e44c27cde
I'm thinking about adding this command to `post-install` scripts for the Linux packages, however, it might really take a long time, like 1-2 minutes so it might not be desired for some users.
